vortex-serde/Cargo.toml

```diff
@@ -25,7 +25,7 @@ futures-util = { workspace = true }
 itertools = { workspace = true }
 lazy_static = { workspace = true }
 monoio = { workspace = true, optional = true, features = ["bytes"] }
-object_store = { workspace = true, optional = true }
+object_store = { workspace = true, optional = true, features = ["aws", "gcp", "azure", "http"] }
```

Comment:
ok, now that I see this we shouldn't have those features here. Which means the url parsing function should likely move out of here.

Reply:
Can you elaborate on why? It seems to me that vortex-serde is, effectively, our file read/write crate. If you want to read/write to object store, you need these features. If you don't need to read/write to object store, you can slim the library size by not enabling the `object_store` optional dependency.

Reply:
You provide use the ObjectStore implementation and we can use it. This crate doesn't have to know about specific implementations and those are only added here because we want to have consistent url parsing. Essentially there exists some package that uses this one and object store to read files

Reply:
i.e. more so this crate will need to be split into two/three separate ones

Reply:
I could pull the URL parser up into PyVortex since PyVortex always gives you the kitchen sink of features.

I could also feature-config the match branches so that the URL Parser only parses the features you enabled. PyVortex would do something like `vortex-serde = { features = ["aws", ...] }`

Reply:
I think we should move it to pyvortex for now

Reply:
Done.